substrate/frame/support/src/traits/tokens/currency.rs

```diff
@@ -16,6 +16,9 @@
 // limitations under the License.
 
 //! The Currency trait and associated types.
+//!
+//! Note Currency and related traits are deprecated, instead
+//! [`fungible`](frame_support::traits::fungible) traits should be used.
```

Comment:
We should actually mark this file as deprecated, possibly in H2.

Reply:
When you say "mark the file as deprecated", do you just mean add a `## Deprecated` to the doc comment?

Reply:
No, I mean actually adding a `#[deprecated]` to `trait Currency`.

Reply:
Created a tracking issue: https://github.com/paritytech/polkadot-sdk/issues/3836